### PR TITLE
Add worker's RQ and Python versions to dashboard

### PR DESCRIPTION
**Pull request by coolhacker**
Have not finished unittest yet

### CODE REVIEW

File: rq_dashboard/templates/rq_dashboard/dashboard.html [hunk 1]
```diff
@@ -75,7 +75,7 @@ <h1>Workers <a id="workers-btn" href="#">(toggle)</a></h1>
             <script name="worker-row" type="text/template">
                 <tr data-role="worker">
                     <td><i class="icon-<%= d.state %>" title="Job ID: <%= d.current_job.job_id %>&#013;&#010;Description: <%= d.current_job.description %>&#013;&#010;Created at: <%= d.current_job.created_at %>&#013;&#010;"></i></td>
-                    <td><%= d.name %></td>
+                    <td title ="RQ <%= d.version %> / Python <%= d.python_version %>"><%= d.name %></td>
```
Comment (eoranged) [inline rq_dashboard/templates/rq_dashboard/dashboard.html]: Please wrap with if.

File: rq_dashboard/web.py [hunk 2]
```diff
@@ -339,9 +339,11 @@ def serialize_queue_names(worker):
             queues=serialize_queue_names(worker),
             state=str(worker.get_state()),
             current_job=serialize_current_job(worker.get_current_job()),
+            version=worker.version,
+            python_version=worker.python_version
```
Comment (eoranged) [inline rq_dashboard/web.py]: Will break with current versions. Please use getattr as discussed.

**Review by eoranged — CHANGES_REQUESTED** [review]

File: rq_dashboard/templates/rq_dashboard/dashboard.html (hunk 1 above)
Comment (coolhacker) [reply rq_dashboard/templates/rq_dashboard/dashboard.html]: Wrapped up.

File: rq_dashboard/web.py (hunk 2 above)
Comment (coolhacker) [reply rq_dashboard/web.py]: Fixed.

**Comment by coolhacker** [comment]
Now everything should be fine, but there may be problems with  unittests
Closes: #210

File: tests/compat.py [hunk 3]
```diff
@@ -14,6 +15,7 @@
 from rq_dashboard.cli import make_flask_app
 
 HTTP_OK = 200
+VALID_RQ_VERSION = '1.2.0'
```
Comment (eoranged) [inline tests/compat.py]: Please rename to VERSION_1_2 or something like this.
Also, please check what will happen if `rq_version.VERSION = '1.10.0'`

File: rq_dashboard/web.py [hunk 4]
```diff
@@ -339,9 +339,11 @@ def serialize_queue_names(worker):
             queues=serialize_queue_names(worker),
             state=str(worker.get_state()),
             current_job=serialize_current_job(worker.get_current_job()),
+            version=getattr(worker, 'version', ''),
+            python_version=getattr(worker, 'python_version','')
```
Comment (eoranged) [inline rq_dashboard/web.py]: Please add space after comma.

**Review by eoranged — CHANGES_REQUESTED** [review]

File: rq_dashboard/web.py (hunk 4 above)
Comment (coolhacker) [reply rq_dashboard/web.py]: Fixed.

File: tests/compat.py (hunk 3 above)
Comment (coolhacker) [reply tests/compat.py]: Reworked.

File: tests/test_basic.py [hunk 5]
```diff
@@ -44,6 +57,24 @@ def test_failed_jobs(self):
...
         self.assertEqual(response.status_code, HTTP_OK)
 
+    def test_worker_python_version_field(self):
+        w = Worker(['q'])
+        w.register_birth()
+        response = self.client.get('/workers.json')
+        data = json.loads(response.data.decode('utf8'))
+        if getattr(w, 'python_version', None):
+            self.assertEqual(w.python_version, data['workers'][0]['python_version'])
+        pop_connection()
```
Comment (eoranged) [inline tests/test_basic.py]: Since It's pushed in setUp, then It will be better to pop It in tearDown.

Comment (coolhacker) [reply tests/test_basic.py]: Yeah, it makes sense. Edited.

File: rq_dashboard/web.py [hunk 6]
```diff
@@ -339,9 +339,11 @@ def serialize_queue_names(worker):
             queues=serialize_queue_names(worker),
             state=str(worker.get_state()),
             current_job=serialize_current_job(worker.get_current_job()),
+            version=getattr(worker, 'version', ''),
+            python_version=getattr(worker, 'python_version', '')
         )
         for worker in Worker.all()),
-        key=lambda w: (w['state'], w['queues'], w['name']))
+        key=lambda w: (w['state'], w['queues'], w['name'], w['version'], w['python_version']))
```
Comment (eoranged) [inline rq_dashboard/web.py]: And why do we need these in sorting?

File: tests/test_basic.py [hunk 7]
```diff
@@ -44,6 +60,22 @@ def test_failed_jobs(self):
         response = self.client.get('/failed')
         self.assertEqual(response.status_code, HTTP_OK)
 
+    def test_worker_python_version_field(self):
+        w = Worker(['q'])
+        w.register_birth()
+        response = self.client.get('/workers.json')
+        data = json.loads(response.data.decode('utf8'))
+        if getattr(w, 'python_version', None):
+            self.assertEqual(w.python_version, data['workers'][0]['python_version'])
```
Comment (eoranged) [inline tests/test_basic.py:67]: It makes sense to add else branch in both tests to standardise behaviour when there is no python/rq version in worker.

File: tests/test_basic.py [hunk 8]
```diff
@@ -3,16 +3,32 @@
 import json
 import unittest
 
+import redis
+
 from rq_dashboard.cli import make_flask_app
 
+from rq import Worker
+from rq import pop_connection, push_connection
```
Comment (eoranged) [inline tests/test_basic.py]: Please apply isort to It.

**Review by eoranged — CHANGES_REQUESTED** [review]
There are a few more things left to polish.
Aside from this, everything looks good.

File: rq_dashboard/web.py (hunk 6 above)
Comment (coolhacker) [reply rq_dashboard/web.py]: You're right, it is unnecessary. Edited.

File: tests/test_basic.py (hunk 7 above)
Comment (coolhacker) [reply tests/test_basic.py:67]: Yes, I don’t know why I missed it. Edited.

File: tests/test_basic.py (hunk 8 above)
Comment (coolhacker) [reply tests/test_basic.py]: Good utility! I did not know about this lib until now. Thanks.

**Review by eoranged — APPROVED** [review]

**Pull request merged**